src/NodeParser/MappedTypeNodeParser.ts

```diff
@@ -24,23 +24,34 @@ export class MappedTypeNodeParser implements SubNodeParser {
         );
     }
 
-    private getObjectProperty(type: any, node: ts.MappedTypeNode, context: Context) {
-        return new ObjectProperty(
-            type.value,
-            this.childNodeParser.createType(node.type!, context),
-            !node.questionToken,
-        );
-    }
-
     private getProperties(node: ts.MappedTypeNode, context: Context): ObjectProperty[] {
-        const type: any = this.typeChecker.getTypeFromTypeNode(node.typeParameter.constraint!);
+        const type: any = this.typeChecker.getTypeFromTypeNode((<any>node.typeParameter.constraint));
 
+        let toRet: ObjectProperty[] = [];
         if (type.types) {
-            return type.types.map((t: any) => this.getObjectProperty(t, node, context));
-        } else if (type.intrinsicName !== "never") {
-            return [this.getObjectProperty(type, node, context)];
-        } else {
-            return [];
+            toRet = type.types.reduce((result: ObjectProperty[], t: any) => {
+                const createdType = this.childNodeParser.createType(node.type!, context);
+
+                const objectProperty = new ObjectProperty(
+                    t.value,
+                    createdType,
+                    !node.questionToken,
+                );
+
+                result.push(objectProperty);
+                return result;
+            }, []);
+        } else if (context.getArguments()) {
+            toRet = context.getArguments().reduce((acc: ObjectProperty[], arg: any) => {
```

Comment:
This reduce can be replaced with a map.

Reply:
I noticed this also - I've got another PR coming through to support pick also so I'll make sure it's in that as map.

Reply:
👍